packages/core/src/Button/Button.tsx

```diff
@@ -130,13 +132,14 @@ export const HvButton = fixedForwardRef(function HvButton<
       className={cx(
         classes.root,
         classes[variant],
-        size && css(getSizeStyles(size)),
+        size && !icon && css(getSizeStyles(size)),
         radius && css(getRadiusStyles(radius)),
         overrideIconColors && css(getOverrideColors()),
         {
           [classes.icon]: icon,
           [classes.disabled]: disabled,
         },
+        size && icon && css(getIconSizeStyles(size)),
```

Comment:
Is this needed?
The purpose of the `icon` prop is so that the Button's styles are unset and the icon (`children`) can decide the size of the button. The `size` styles come before `classes.icon` because of that.
If we need a larger icon button, we should increase the icon size
Am I missing something? 🤔

Reply:
I made this change so that it's possible for a icon button to have different sizes and still have the 'squarish' look like in the [specs](https://designsystem.hitachivantara.com/6c705d900/p/73880a-button/b/50959e) even when the icon is smaller. It was also useful for integrating an icon button like the dropmenu with different sizes in a `Multibutton` and have them matching the height. But let me know if you see a better way to do this